### PR TITLE
reduce excess code

### PR DESCRIPTION
**Pull request by Jeremy-Run**
If a key exists in the Index of DB, then `record.Type` can't be `LogRecordDeleted`, otherwise I suggest deleting it from the index.

### CODE REVIEW

File: batch.go [hunk 1]
```diff
@@ -142,6 +140,8 @@ func (b *Batch) Delete(key []byte) error {
 	}
 
 	b.mu.Lock()
+	defer b.mu.Unlock()
```
Comment (roseduan) [inline batch.go:143]: no need to use defer since the following code is a very common operation without any errors.

File: batch.go [hunk 2]
```diff
@@ -121,11 +122,8 @@ func (b *Batch) Get(key []byte) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-
 	record := decodeLogRecord(chunk)
-	if record.Type == LogRecordDeleted {
```
Comment (roseduan) [inline batch.go:126]: we can add an assertion here if we assume that the Type will never be `LogRecordDeleted`.
eg. if record.Type == LogRecordDeleted { panic("some prompt messages...")}

**Comment by Jeremy-Run** [comment]
@roseduan Do you think it's a good idea to just delete it from the index?

**Comment by roseduan** [comment]
> @roseduan Do you think it's a good idea to just delete it from the index?

I guess no, because we assume it will never happen, so an assert seems reasonable.

**Comment by Jeremy-Run** [comment]
> > @roseduan Do you think it's a good idea to just delete it from the index?
> 
> I guess no, because we assume it will never happen, so an assert seems reasonable.

**Comment by Jeremy-Run** [comment]
I accidentally closed it. 😂

**Pull request closed without merging**